website/docs/r/projectRole.html.markdown

```diff
@@ -36,6 +36,8 @@ The following arguments are supported:
 * `annotations` - (Optional/Computed) Annotations of the resource (map)
 * `labels` - (Optional/Computed) Labels of the resource (map)
 
+**Note** user `user_id | user_principal_id` OR group `group_id | group_principal_id` must be defined
```

Comment:
When would you use user_id or group_id over the principal version?

Reply:
Probably never, but there is the option at API.. :)